Alec/Recording and playback page API Implementation

Alec-Hallman opened the pull request:
This Pr adds:
- Pushes recordings into the API in the stop recording function of the recording page
- In the playback page it now calls the API to get the transcript, emotions, and title.
- Stats page now calls the Hive box to fetch every single recording, checks the recording's date, splits those into different lists and then those list elements are counted to be used by the page. This is a pretty inefficient way of doing this sorry in advance but it works!
- Added a delete button to the playback page so entries can now be deleted from the database (I'm pretty sure they still exist on the devices storage though)

Alec-Hallman committed:
Merge branch 'rosemary/api-return-top-emotions' into FrontendAPI

Alec-Hallman committed:
Frontend API Testing

Merged my branch and Rosemary's API branch

Alec-Hallman committed:
Merge branch 'rosemary/api-return-top-emotions' into FrontendAPI

Alec-Hallman committed:
Base API code (not working)

Alec-Hallman committed:
Front end call working and foundation of transcript fetching

Alec-Hallman committed:
API implementation

Alec-Hallman committed:
Merge branch 'main' into FrontendAPI

Alec-Hallman committed:
Merge from main, big fixes and adding emotional updating

Alec-Hallman committed:
Stats page foundation

Alec-Hallman committed:
Stats page implementation

Alec-Hallman committed:
Fixed a bug where emotions were being repeatedly counted

Alec-Hallman committed:
Added a delete recording button to the playback page

You can now delete entries from the database.

anikapatychuk reviewed (approved):
Looks good to me! I see what you mean with the recordings but given the time crunch I think its ok

Alec-Hallman committed:
Scrollable transcript. Gradient recording colours and statistics page which fetches transcript from un opened recordings

Alec-Hallman committed:
Added home page update on enter from bottom navigation bar

Alec-Hallman committed:
Removed data base calls

My API call was inside by database fetch so that's missing atm. I'll add it to the page again once this branch gets approved so I can re pull from main and add it implement it in a way I can test.

pabloj2001 committed:
Merge branch 'main' into FrontendAPI

Alec-Hallman committed:
Fixing some merge issues

-Fixed box error in home page
- Deleted the repeated Emotion stats box

pabloj2001 committed:
Updated app colors and home recordings list margins
